scalyr_agent/builtin_monitors/syslog_utils.py

```diff
@@ -0,0 +1,211 @@
+
+    def _get_old_logs_for_glob(
+        self, current_time, glob_pattern, existing_logs, check_rotated, max_rotations
+    ):
+
+        result = []
+
+        for matching_file in glob.glob(glob_pattern):
+            try:
+                added = False
+                mtime = os.path.getmtime(matching_file)
+                if (
+                    current_time - mtime > self._delete_interval
+                    and matching_file not in existing_logs
+                ):
+                    result.append(matching_file)
+                    added = True
+
+                for i in range(max_rotations, 0, -1):
+                    rotated_file = matching_file + (".%d" % i)
+                    try:
+                        if not os.path.isfile(rotated_file):
+                            continue
+
+                        if check_rotated:
+                            mtime = os.path.getmtime(rotated_file)
+                            if current_time - mtime > self._delete_interval:
+                                result.append(rotated_file)
+                        else:
+                            if added:
+                                result.append(rotated_file)
+
+                    except OSError as e:
+                        global_log.warn(
+                            "Unable to read modification time for file '%s', %s"
+                            % (rotated_file, six.text_type(e)),
+                            limit_once_per_x_secs=300,
+                            limit_key="mtime-%s" % rotated_file,
+                        )
+
+            except OSError as e:
+                global_log.warn(
+                    "Unable to read modification time for file '%s', %s"
+                    % (matching_file, six.text_type(e)),
+                    limit_once_per_x_secs=300,
+                    limit_key="mtime-%s" % matching_file,
+                )
+        return result
+
+    def check_for_old_logs(self):
+        old_logs = []
+        current_time = time.time()
+        if current_time - self._last_check > self._check_interval:
+
+            old_logs = self._get_old_logs_for_glob(
+                current_time,
+                self._log_glob,
+                self.existing_logs,
+                self._check_rotated_timestamps,
+                self._max_log_rotations,
+            )
+            self._last_check = current_time
+
+        for filename in old_logs:
+            try:
+                os.remove(filename)
+                global_log.log(
+                    scalyr_logging.DEBUG_LEVEL_1, "Deleted old log file '%s'" % filename
+                )
+            except OSError as e:
+                global_log.warn(
+                    "Error deleting old log file '%s', %s"
+                    % (filename, six.text_type(e)),
+                    limit_once_per_x_secs=300,
+                    limit_key="delete-%s" % filename,
+                )
+
+        self.existing_logs = []
```

Comment:
Not sure if I understand why we need to reset this variable here.

Reply:
This is something I attempted to bring over from the old implementation of LogDeleter, my understanding of the old implementation is that this list gets populated by logs that have been written to recently (since the last time we checked for old logs) to avoid checking them for deletion. If we didn't clear it nothing would ever get deleted, and we clear it here because to set all logs to once again be able to be deleted.

Having said that, I have no idea why we really need this. If a log was written recently we would find a fresh write time and they wouldn't get deleted anyway. If you see any reason to keep this please let me know, but if you can't find any reasoning behind it either then I would be happy to remove it.